### DISCUSSION TITLE
the pitch angle

### DISCUSSION THREAD

**Discussion by yinghanguan** [NatLabRockies/floris#855] [Q&A]
In floris, what parameter can be controlled to change the pitch angle of the turbine and hence its active power?
In which subfile is this control parameter located？

**Comment by misi9170** [comment]
Hi @yinghanguan , I've responded to you over on #852 , but the way to "control" the blade pitch angle is with a change to the turbine power and thrust curves (as @CaiZhiming-HHU pointed out). If you know the effect that the change in pitch angle has on the turbine power and turbine thrust, you can simply provide updated curves within the turbine yaml file (by changing the `power_thrust_table` values, e.g. on the [NREL 5MW model](https://github.com/NREL/floris/blob/main/floris/turbine_library/nrel_5MW.yaml#L43)). 

This already is done for above rated operation, where the turbine begins pitching the blades to reduce the aerodynamic torque, thereby also reducing the power coefficient and thrust to maintain rated power. 

If you would like to specify various different power and thrust curves for the turbine at different blade pitch angles, consider using the mutlidimensional turbine capability (#711 ) to specify curves at fixed blade pitch angles.

Finally, I'll note that in [Floris v4](https://github.com/NREL/floris/tree/v4), which will be released in the next few weeks as the latest version of Floris, users can [specify `power_setpoints`](https://github.com/NREL/floris/blob/v4/floris/floris_model.py#L308) to limit the rated power, which essentially allows control over the active power of the turbine. This was introduced in #783.
